### PR TITLE
Deploy target

### PR DESCRIPTION
**Pull request by chainlink**

### CODE REVIEW

**Comment by chainlink** [comment]
Closes: https://github.com/pachyderm/helmchart/issues/162

**Comment by chainlink** [comment]
Still a bit more to do here (cleaning up google storage classes) but largely ready for review!

File: pachyderm/templates/etcd/statefulset.yaml [hunk 1]
```diff
@@ -80,7 +80,7 @@ spec:
       name: etcd-storage
       namespace: {{ .Release.Namespace }}
     spec:
-      {{- if not (eq .Values.pachd.storage.backend "LOCAL") }}
+      {{- if or .Values.etcd.storageClass  (eq .Values.deployTarget "GOOGLE") (eq .Values.deployTarget "MICROSOFT") (eq .Values.deployTarget "AMAZON") }}
```
Comment (robert-uhl) [inline pachyderm/templates/etcd/statefulset.yaml:83]: Would it be wrong to say `{{- if or .Values.etcd.storageClass (not eq .Values.deployTarget "LOCAL") }}`?

File: pachyderm/values.yaml [hunk 2]
```diff
@@ -7,6 +7,11 @@
 # In accordance with pachctl deploy it is _not_ per-image.
 imagePullSecret: ""
 
+# Deploy Target configures the storage backend to use and cloud provider 
+# settings (storage classes, etc). It must be one of GOOGLE, AMAZON,
```
Comment (robert-uhl) [inline pachyderm/values.yaml:11]: ‘If set, it must be one of …’

File: pachyderm/values.yaml [hunk 3]
```diff
@@ -153,6 +158,7 @@ pachd:
     additionalAnnotations: {}
     name: "pachyderm" #TODO Set default in helpers / Wire up in templates
   storage:
+    backend: ""
```
Comment (robert-uhl) [inline pachyderm/values.yaml:161]: Moderate style not: missing documentation.

Minor style nit: properties not alphabetically sorted.

File: test/google_test.go [hunk 4]
```diff
@@ -89,12 +89,12 @@ func TestGoogle(t *testing.T) {
...
-					if sa := resource.Annotations["iam.gke.io/gcp-service-account"]; sa != expectedServiceAccount {
-						t.Errorf("expected service account to be %q but was %q", expectedServiceAccount, sa)
-					}
-				})
+				/*
+					t.Run(fmt.Sprintf("%s service account annotation equals %s", resource.Name, expectedServiceAccount), func(t *testing.T) {
+						if sa := resource.Annotations["iam.gke.io/gcp-service-account"]; sa != expectedServiceAccount {
+							t.Errorf("expected service account to be %q but was %q", expectedServiceAccount, sa)
+						}
+					})*/
```
Comment (robert-uhl) [inline test/google_test.go:97]: Should probably refactor or delete rather than leave commented-out.

**Review by robert-uhl — APPROVED** [review]
Couple of nits, but otherwise I like it: adding another level of indirection for deployment in this case is IMHO a win.

File: pachyderm/templates/etcd/statefulset.yaml (hunk 1 above)
Comment (chainlink) [reply pachyderm/templates/etcd/statefulset.yaml:83]: So, I wanted to make it explicit incase we add more cloud providers (and to support the "Custom" deploy target)

File: pachyderm/values.yaml (hunk 3 above)
Comment (chainlink) [reply pachyderm/values.yaml:164]: Added docs, I put it at the beginning as it's kind of a special value

**Pull request merged**